### PR TITLE
MediaDevices.produceCropTarget -> CropTarget.fromElement

### PR DESCRIPTION
**Pull request by eladalon1983**
Change the point of exposure of token-minting from:
  MediaDevices.produceCropTarget()
To:
  CropTarget.fromElement()

Everything else is kept as-is, to be debated and improved in subsequent PRs.

***
<a href="https://pr-preview.s3.amazonaws.com/eladalon1983/mediacapture-region/pull/50.html" title="Last updated on May 31, 2022, 8:59 AM UTC (82c3ec7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-region/50/1019828...eladalon1983:82c3ec7.html" title="Last updated on May 31, 2022, 8:59 AM UTC (82c3ec7)">Diff</a>

### CODE REVIEW

**Comment by eladalon1983** [comment]
@jan-ivar has requested that Chrome should hold off on shipping Region Capture so that we may keep discussing. Both @jan-ivar and @youennf have been extremely responsive in interacting with the discussion on the various issues of this API in the last few weeks. I would be gratified if you could also be similarly responsive in reviewing PRs that solidify the consensus we have worked so hard to achieve.

CC @dontcallmedom

File: index.html
Comment (jan-ivar) [inline index.html:169]: With an element

Comment (jan-ivar) [reply index.html:169]: I'd also remove "supported type" as that seems vague, but I appreciate the code move is causing the diff to grow, and would happily deal with that in a followup as well.

File: index.html
Comment (jan-ivar) [inline index.html:180]: With a given element

File: index.html
Comment (jan-ivar) [inline index.html:196]: The note reads wrong. Please match https://github.com/w3c/mediacapture-region/issues/11#issuecomment-1129883808

**Review by jan-ivar — CHANGES_REQUESTED** [review]

File: index.html
Comment (eladalon1983) [reply index.html]: Done.

File: index.html
Comment (eladalon1983) [reply index.html]: Done.

File: index.html
Comment (beaufortfrancois) [inline index.html:151]: The text above should be edited as well. It currently says:

```
          CropTarget is an intentionally empty, opaque identifier that exposes nothing. Its sole
          purpose is to be handed to {{BrowserCaptureMediaStreamTrack/cropTo}} as input.
```

File: index.html
Comment (beaufortfrancois) [inline index.html:197]:
```suggestion
                or using a synchronous constructor that accepts an {{Element}} as
```

**Review by beaufortfrancois — COMMENTED** [review]
I believe the sample code at https://w3c.github.io/mediacapture-region/#sample-code should be using `CropTarget.fromElement()` instead of `mediaDevices.produceCropTarget()`.

The README.md file would also benefit from these changes for consistency

File: index.html
Comment (eladalon1983) [reply index.html:197]: Thanks; I'll upload that edit from my local machine (it lets me run the formatter).

File: index.html
Comment (beaufortfrancois) [reply index.html:151]: The "exposes nothing" part was what triggered me as `CropTarget.fromElement()` is something to me.
Your call though ;)

Comment (eladalon1983) [reply index.html:151]: I could remove "exposes nothing". Would that make things clearer?

Comment (eladalon1983) [reply index.html:151]: Done.

Comment (beaufortfrancois) [reply index.html:151]: LGTM

File: index.html
Comment (beaufortfrancois) [inline index.html:154]: We may want to add a note explaining why `[SecureContext]` is not on the interface.

**Review by beaufortfrancois — APPROVED** [review]

File: index.html
Comment (jan-ivar) [inline index.html:172]: This doesn't belong here which I've stated in https://github.com/w3c/mediacapture-region/issues/29#issuecomment-1074008247, but since this is in some ways a code-move, I'll accept a separate PR to remove it, in order to break down changes into smaller PRs.

File: index.html
Comment (jan-ivar) [inline index.html:197]: No such thing as an asynchronous constructor afaik.
```suggestion
                or a {{CropTarget}} constructor that accepts an {{Element}} as input. This is
```

File: index.html
Comment (jan-ivar) [inline index.html:195]: Might it be simpler to say
```suggestion
                There is no consensus yet on whether producing a {{CropTarget}}
```
?

**Review by jan-ivar — CHANGES_REQUESTED** [review]

File: index.html
Comment (eladalon1983) [reply index.html:169]: This is prose was moved unchanged. The status quo of the document has not been altered. Let's have a separate discussion for this.

File: index.html
Comment (eladalon1983) [reply index.html:172]: This is prose was moved unchanged. The status quo of the document has not been altered. Let's have a separate discussion for this.

File: index.html
Comment (eladalon1983) [reply index.html:197]: Committed.

**Comment by youennf** [comment]
I made this API proposal based on the consensus we achieved that all elements would be supported.
Given https://github.com/w3c/mediacapture-region/issues/51 wants to revisit this, this makes me wonder whether we should pause this PR until we resolve https://github.com/w3c/mediacapture-region/issues/51.

**Comment by eladalon1983** [comment]
> I made this API proposal based on the consensus we achieved that all elements would be supported. Given #51 wants to revisit this, this makes me wonder whether we should pause this PR until we resolve #51.

The current PR would still be an improvement over `mediaDevices.produceCropTarget()` and we should therefore merge it. Also, please note this message I had posted on #51:

> @youennf, you have previously agreed HTMLElement made sense, and https://github.com/w3c/mediacapture-region/issues/13, we even expanded to Element. You made this suggestion five months ago. The API as a whole has been debated for 1.5 years by now. I do not think this qualifies as "rushed". It is the nature of our jobs that decisions must be made without full certainty. Past a certain point, additional debates are less productive than making a decision, even if that decision could end up proving less than perfectly optimal. (To clarify - I think we're making a good decision in https://github.com/w3c/mediacapture-region/pull/50.)

Let's press onwards. Many more productive issues await us.

**Comment by eladalon1983** [comment]
@jan-ivar and @youennf, PTAL.

**Review by jan-ivar — APPROVED** [review]

**Pull request merged**
